Automattic/jetpack
# Publicize: adjust feature description depending on the state

### Branched off from https://github.com/Automattic/jetpack/pull/21366

This PR adjusts the main description and the enable/disable connection states of the feature depending on its feature flag and main sharing switch (main toggle control).

Fixes #

#### Changes proposed in this Pull Request:

* RePublicize: adjust feature description depending on the state

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Test with Jetpack plugin activated
* Enable Sharing Jetpack module
* Confirm the feature description depending on the feature flag, and the main toggle control

### Feature Flag disabled
--------------
#### New post / No connections
<img src="https://user-images.githubusercontent.com/77539/136861523-e74bfe0b-c8d4-47c3-ae38-32cce0a03453.png" width="300px" />

--------------
#### New post / Connections available
<img src="https://user-images.githubusercontent.com/77539/136925415-f90939a7-aa9b-41d0-916d-fc642d3a75ce.png" width="300px" />

----------
#### Published post / No connections
<img src="https://user-images.githubusercontent.com/77539/136861869-1f4fbb59-6e65-48a0-b5f7-85f0ca7ada3e.png" width="300px" />

----------
#### Published post / Connections available
<img src="https://user-images.githubusercontent.com/77539/136925717-044c0f43-e094-460a-815c-fba28351e5b1.png" width="300px" />

--------------

### Feature flag enabled

Enabling instructions in [this issue](https://github.com/Automattic/jetpack/pull/21366)

--------

#### New post / Sharing disabled / No connections
<img src="https://user-images.githubusercontent.com/77539/136930769-631d66d1-6aa8-4b2b-9691-34d615253f28.png" width="300px" />

--------

#### New post / Sharing disabled / Connections available
<img src="https://user-images.githubusercontent.com/77539/136928788-f59d6bb3-469e-44d9-aff2-484ff0a447ea.png" width="300px" />

---------

#### New post / Sharing enabled / No connections enabled
<img src="https://user-images.githubusercontent.com/77539/136932096-17f0b0c8-38cf-42ed-a71c-de54459d9810.png" width="300px" />

-----------

#### New post / Sharing enabled / Some connection enabled
<img src="https://user-images.githubusercontent.com/77539/136932281-e11b980c-f418-46e6-b8c3-45d90e3310c8.png" width="300px" />

-----------

#### Published post / Sharing disabled / No connections
<img src="https://user-images.githubusercontent.com/77539/136931649-3fb75922-d7be-4f4b-98ca-875ce5270000.png" width="300px" />

-------------

#### Published post / Sharing disabled / connections available
<img src="https://user-images.githubusercontent.com/77539/136929001-f66e2dff-ff21-4bd6-84a3-bb3dafd124d9.png" width="300px" />

-----------

#### Published post / Sharing enabled / Some connection enabled

<img src="https://user-images.githubusercontent.com/77539/136932955-56751e89-a32e-43c6-ac3a-01d5e8e58c3d.png" width="300px" />

_we'll unblock the toggle button when RePublicize is enabled in a follow-up PR_
